src/main/steps/applicant1/hub-page/joint/content.ts

```diff
@@ -306,54 +308,41 @@ export const generateContent: TranslationFn = content => {
-
   const displayState = currentStateFn(userCase.state).at(
     (userCase.state === State.OfflineDocumentReceived ? userCase.previousState : userCase.state) as State
   );
 
   const finalOrderEligibleAndSecondInTimeFinalOrderNotSubmittedWithin14Days =
-    hasApplicantAppliedForFinalOrderFirst &&
+    hasApplicantAppliedForFoFirst(userCase, isApplicant2) &&
     dayjs().isBefore(userCase.dateFinalOrderNoLongerEligible) &&
     dayjs().isAfter(
       dayjs(userCase.dateFinalOrderSubmitted).add(config.get('dates.finalOrderSubmittedOffsetDays'), 'day')
     ) &&
     userCase.state === State.AwaitingJointFinalOrder;
 
-  const doesApplicantIntendToSwitchToSoleFinalOrder = isApplicant2
-    ? userCase.doesApplicant2IntendToSwitchToSole === YesOrNo.YES
-    : userCase.doesApplicant1IntendToSwitchToSole === YesOrNo.YES;
-
-  const dateApplicantDeclaredIntentionToSwitchToSoleFo = isApplicant2
-    ? userCase.dateApplicant2DeclaredIntentionToSwitchToSoleFo
-    : userCase.dateApplicant1DeclaredIntentionToSwitchToSoleFo;
+  const isFinalOrderAwaitingOrOverdue =
+    displayState.state() === State.AwaitingFinalOrder ||
+    displayState.state() === State.AwaitingJointFinalOrder ||
+    displayState.state() === State.FinalOrderOverdue;
 
-  const withinSwitchToSoleFinalOrderIntentionNotificationWindow =
-    hasApplicantAppliedForFinalOrderFirst &&
-    doesApplicantIntendToSwitchToSoleFinalOrder &&
-    dayjs().isAfter(dateApplicantDeclaredIntentionToSwitchToSoleFo) &&
-    dayjs().isBefore(
-      dayjs(dateApplicantDeclaredIntentionToSwitchToSoleFo).add(
-        config.get('dates.switchToSoleFinalOrderIntentionNotificationOffsetDays'),
-        'day'
-      )
-    ) &&
-    userCase.state === State.AwaitingJointFinalOrder;
+  const isFinalOrderAwaitingOrOverdueAndApplicantHasNotAppliedFirst =
+    isFinalOrderAwaitingOrOverdue && !hasApplicantAppliedForFoFirst(userCase, isApplicant2);
 
   const applicantConfirmReceipt = isApplicant2 ? 'applicant2ConfirmReceipt' : 'applicant1ConfirmReceipt';
   const applicantApplyForConditionalOrderStarted = isApplicant2
     ? 'applicant2ApplyForConditionalOrderStarted'
     : 'applicant1ApplyForConditionalOrderStarted';
 
+  const switchToSoleFinalOrderStatus = getSwitchToSoleFoStatus(userCase, isApplicant2);
+
   const isFinalOrderCompleteState = userCase.state === State.FinalOrderComplete;
-  const theLatestUpdateTemplate = getJointHubTemplate(
-    displayState,
-    userCase,
+  const theLatestUpdateTemplate = getJointHubTemplate(displayState, userCase, {
     hasApplicantAppliedForConditionalOrder,
-    withinSwitchToSoleFinalOrderIntentionNotificationWindow
-  );
+    isWithinSwitchToSoleFoIntentionNotificationPeriod:
+      switchToSoleFinalOrderStatus.isWithinSwitchToSoleFoIntentionNotificationPeriod,
+    hasSwitchToSoleFoIntentionNotificationPeriodExpired:
+      switchToSoleFinalOrderStatus.hasSwitchToSoleFoIntentionNotificationPeriodExpired,
+  });
 
```

Comment:
why not just pass switchToSoleFinalOrderStatus, this seems to be recreating the same object

Reply:
I was working on the principle of only passing the properties that are required.